lib/matplotlib/quiver.py

```diff
@@ -542,7 +542,36 @@ def draw(self, renderer):
         super().draw(renderer)
         self.stale = False
 
+    def set_U(self, U):
+        """Set x direction components of the arrow vectors."""
+        self.set_UVC(U, None, None)
+
+    def set_V(self, V):
+        """Set y direction components of the arrow vectors."""
+        self.set_UVC(None, V, None)
+
+    def set_C(self, C):
+        """Set the arrow colors."""
+        self.set_UVC(None, None, C)
+
     def set_UVC(self, U, V, C=None):
+        """
+        Set the U, V (x and y direction components of the arrow vectors) and
+        C (arrow colors) values of the arrows.
+
+        Parameters
+        ----------
+        U : ArrayLike | None
+            The x direction components of the arrows. If None it is unchanged.
+        V : ArrayLike | None
+            The y direction components of the arrows. If None it is unchanged.
+        C : ArrayLike | None, optional
+            The arrow colors. The default is None.
+        """
+        if U is None:
+            U = self.U
+        if V is None:
+            V = self.V
         # We need to ensure we have a copy, not a reference
         # to an array that might change before draw().
         U = ma.masked_invalid(U, copy=True).ravel()
```

Comment:
If I have understood, this could be in an `else` branch of the `if U is None:` loop above.

Reply:
It is not possible because of the mask.